packages/salesforcedx-vscode-core/src/commands/util/postconditionCheckers.ts

```diff
@@ -55,3 +65,141 @@ export class FilePathExistsChecker
     return overwrite === nls.localize('warning_prompt_continue_confirm');
   }
 }
+
+/* tslint:disable-next-line:prefer-for-of */
+export class OverwriteComponentPrompt
+  implements PostconditionChecker<OneOrMany> {
+  public async check(inputs: ContinueOrCancel): Promise<ContinueOrCancel> {
+    if (inputs.type === 'CONTINUE') {
+      const { data } = inputs;
+      // normalize data into a list when processing
+      const componentsToCheck = data instanceof Array ? data : [data];
+      const foundComponents = componentsToCheck.filter(component =>
+        this.componentExists(component)
+      );
+      if (foundComponents.length > 0) {
+        const toSkip = await this.promptOverwrite(foundComponents);
+        // cancel command if cancel clicked or if skipping every file to be retrieved
+        if (!toSkip || toSkip.size === componentsToCheck.length) {
+          return { type: 'CANCEL' };
+        }
+        if (data instanceof Array) {
+          inputs.data = componentsToCheck.filter(
+            selection => !toSkip.has(selection)
+          );
+        }
+      }
+      return inputs;
+    }
+    return { type: 'CANCEL' };
+  }
+
+  private componentExists(component: LocalComponent) {
+    const { fileName, type, outputdir } = component;
+    const info = MetadataDictionary.getInfo(type);
+    const pathStrategy = info
+      ? info.pathStrategy
+      : PathStrategyFactory.createDefaultStrategy();
+    return this.getFileExtensions(component).some(extension => {
+      const path = join(
+        getRootWorkspacePath(),
+        pathStrategy.getPathToSource(outputdir, fileName, extension)
+      );
+      return existsSync(path);
+    });
+  }
+
+  private getFileExtensions(component: LocalComponent) {
+    const info = MetadataDictionary.getInfo(component.type);
```

Comment:
You can pass `info` as a parameter since you've already called it as part of `componentExists`.

Reply:
I would do that, although I need both the `suffix` and `type` properties off of `component`